code/core/src/manager/components/mobile/navigation/MobileMenuDrawer.tsx

```diff
@@ -1,44 +1,59 @@
 import type { FC } from 'react';
-import React, { useRef } from 'react';
+import React, { useCallback, useRef } from 'react';
 
 import { Transition } from 'react-transition-group';
 import type { TransitionStatus } from 'react-transition-group/Transition';
 import { styled } from 'storybook/theming';
 
 import { MOBILE_TRANSITION_DURATION } from '../../../constants';
+import { useModalDialog } from '../../../hooks/useModalDialog';
 import { useLayout } from '../../layout/LayoutProvider';
 import { MobileAbout } from '../about/MobileAbout';
 
 interface MobileMenuDrawerProps {
   children?: React.ReactNode;
+  id?: string;
 }
 
-export const MobileMenuDrawer: FC<MobileMenuDrawerProps> = ({ children }) => {
-  const containerRef = useRef<HTMLDivElement>(null);
+export const MobileMenuDrawer: FC<MobileMenuDrawerProps> = ({ children, id }) => {
   const sidebarRef = useRef<HTMLDivElement>(null);
   const overlayRef = useRef<HTMLDivElement>(null);
   const { isMobileMenuOpen, setMobileMenuOpen, isMobileAboutOpen, setMobileAboutOpen } =
     useLayout();
 
+  const handleClose = useCallback(() => {
+    setMobileMenuOpen(false);
+  }, [setMobileMenuOpen]);
+
+  const dialogRef = useModalDialog({
+    isOpen: isMobileMenuOpen,
+    onClose: handleClose,
+  });
+
   return (
     <>
       <Transition
-        nodeRef={containerRef}
+        nodeRef={dialogRef}
         in={isMobileMenuOpen}
         timeout={MOBILE_TRANSITION_DURATION}
         mountOnEnter
         unmountOnExit
-        onExited={() => setMobileAboutOpen(false)}
+        onExited={() => {
+          setMobileAboutOpen(false);
+          if (dialogRef.current && dialogRef.current.hasAttribute('open')) {
+            dialogRef.current.close();
+          }
+        }}
       >
```

Comment:
Not sure what's going on here, but the exit animation is broken nonetheless. I need to brush up on React transitions to be of help here, but the `close()` call from native dialog is immediate, so it probably should happen at the end of the exit transition, with another layer of state management to handle triggering the exit transition.

Let's see what the people who have written React code in the past 3 years say, because I'm very rusty :D

Reply:
This seems to be fixed now, resolving.